teamd/teamd_per_port.c

```diff
@@ -217,7 +217,6 @@ static void port_obj_remove(struct teamd_context *ctx,
 	struct teamd_port *tdport = _port(port_obj);
 
 	teamd_event_port_removed(ctx, tdport);
-	teamd_port_remove(ctx, tdport);
```

Comment:
I would suggest changing this to:

```
if (!ctx->no_quit_destroy)
	teamd_port_remove(ctx, tdport
```

...which is more in line with the commit message.